demos/matrix_multiply.rst

```diff
@@ -0,0 +1,100 @@
+
+
+Matrix Multiply Host Script
+===========================
+
+The host is also responsible for loading the firmware containing the :ref:`RPmsg Matrix Multiply Remote Application<matrix-multiply-remote-app>` and starting the remote processor using :ref:`Remoteproc<overview-remoteproc-work-label>`.
+
+For host controllers, like Linux, a script can be used to pipe the firmware to the exposed remoteproc system, followed by the execution of the user space mat_mul_demo application. For controllers without scripting capability, like baremetal and RTOS (Real Time Operating systems), this would be achieved in the code.
+
+**********************
+Matrix Multiply Source
+**********************
+
+RPmsg Matrix Multiply Baremetal Sources
+=======================================
+
+There are two baremetal applications, a daemon/service to run on the remote and a host/controller application which is the matrix multiply client requesting the calculations.
+
+The RPmsg Matrix Multiply daemon/service application is available as a baremetal solution in the `OpenAMP Repository <https://github.com/OpenAMP/open-amp/blob/main/apps/examples/matrix_multiply/matrix_multiplyd.c>`_. Take note of the d for daemon at the end of the file.
+
+The RPmsg Matrix Multiple host client application is available as a baremetal solution in the `OpenAMP Repository <https://github.com/OpenAMP/open-amp/blob/main/apps/examples/matrix_multiply/matrix_multiply.c>`_
+
+Both are CMake application and can be built for any remote as long as the relevant :ref:`OS/HW abstraction layer<porting-guide-work-label>` components like libmetal are ported for that platform.
+
+.. _matrix-multiply-linux-app:
+
+Matrix Multiply Linux Source
+============================
+
+The matrix multiply Linux application is executed on the Linux host controller as a user space application.
+The application is available in the `OpenAMP System Reference repository <https://github.com/OpenAMP/openamp-system-reference/blob/main/examples/linux/rpmsg-mat-mul/mat_mul_demo.c>`_.
+
+It is a Makefile application and can be built using the `Yocto rpmsg-mat-mul recipe <https://github.com/OpenAMP/meta-openamp/blob/master/recipes-openamp/rpmsg-examples/rpmsg-mat-mul_1.0.bb>`_
+
+An example host control script is given in the `matrix multiply readme <https://github.com/OpenAMP/openamp-system-reference/tree/main/examples/linux/rpmsg-mat-mul#run-the-demo>`_
+
+*******************************
+Reference Board Implementations
+*******************************
+
+This Matrix Multiply Sample is demonstrated in the following reference implementations.
+
+* :ref:`Docker Images<docker-images-label>` as demo1B
+* :ref:`AMD-Xilinx platforms<demos-AMD--work-label>`
```

Comment:
same it is available using Linux processes

Reply:
done